tst/EnergyPlus/unit/CurveManager.unit.cc

```diff
@@ -742,3 +744,233 @@ TEST_F(EnergyPlusFixture, CSV_CarriageReturns_Handling)
+}
+
+INSTANTIATE_TEST_SUITE_P(CurveManager,
+                         OutputUnitTypeIsValid,
+                         testing::Values("", "Capacity", "Dimensionless", "Power", "Pressure", "Temperature"),
+                         [](const testing::TestParamInfo<InputUnitTypeIsValid::ParamType> &info) -> std::string {
+                             if (info.param.empty()) {
+                                 return "Blank";
+                             }
+                             return std::string{info.param};
+                         });
+
+std::pair<std::set<std::string>, std::set<std::string>> getAllPossibleInputOutputTypesForCurves()
+{
+    const auto json_schema = nlohmann::json::from_cbor(EmbeddedEpJSONSchema::embeddedEpJSONSchema());
+    auto const &schema_properties = json_schema.at("properties");
+    std::set<std::string> all_input_choices;
+    std::set<std::string> all_output_choices;
+
+    for (const auto &[objectType, object_schema] : schema_properties.items()) {
+        const bool is_curve = (objectType.rfind("Curve:", 0) == 0) || (objectType == "Table:Lookup") || (objectType == "Table:IndependentVariable");
+        if (!is_curve) {
+            continue;
+        }
+        auto const &schema_obj_props = getPatternProperties(object_schema);
+        for (const auto &[fieldName, schema_field_obj] : schema_obj_props.items()) {
+            if (std::string(fieldName) == "output_unit_type") {
+                for (const auto &e : schema_field_obj.at("enum")) {
+                    all_output_choices.insert(std::string{e});
+                }
+            } else if (fieldName.find("unit_type") != std::string::npos) {
+                for (const auto &e : schema_field_obj.at("enum")) {
+                    all_input_choices.insert(std::string{e});
+                }
+            }
+        }
+    }
+
+    return {all_input_choices, all_output_choices};
+}
+
+TEST_F(EnergyPlusFixture, AllPossibleUnitTypeValid)
+{
+    auto const [all_input_choices, all_output_choices] = getAllPossibleInputOutputTypesForCurves();
+
+    // As of 2024-12-18
+    // in = ["", "Angle", "Dimensionless", "Distance", "MassFlow", "Power", "Pressure", "Temperature", "VolumetricFlow","VolumetricFlowPerPower"]
+    // out = ["", "Capacity", "Dimensionless", "Power", "Pressure", "Temperature"]
+    EXPECT_FALSE(all_input_choices.empty()) << fmt::format("{}", all_input_choices);
+    EXPECT_FALSE(all_output_choices.empty()) << fmt::format("{}", all_output_choices);
+
+    for (const auto &input_unit_type : all_input_choices) {
+        EXPECT_TRUE(Curve::IsCurveInputTypeValid(input_unit_type)) << input_unit_type << " is rejected by IsCurveOutputTypeValid";
+    }
+
+    for (const auto &output_unit_type : all_output_choices) {
+        if (output_unit_type.empty()) {
+            continue;
+        }
+        EXPECT_TRUE(Curve::IsCurveOutputTypeValid(output_unit_type)) << output_unit_type << " is rejected by IsCurveOutputTypeValid";
+    }
+}
```

Comment:
Dynamic test that collates ALL possible input/output unit types by mining the schema for all curve objects and tests that IsCurve<In/Out>putTypeValid is ok. It wasn't for VolumetricFlowPerPower